### ISSUE TITLE
Inconsistent treatment of `role` argument in IAM

### ISSUE THREAD

**Issue by mmdriley** [pulumi/pulumi-aws#30]
In this example, I want to create an IAM role, attach an inline policy, then create an EC2 instance profile with that role.

```typescript
let assumeInstanceRolePolicyDoc = // ...
let instanceRolePolicyDoc = // ...

let instanceRole = new aws.iam.Role("ecsClusterInstanceRole", {
    assumeRolePolicy: JSON.stringify(assumeInstanceRolePolicyDoc),
});

let instanceRolePolicy = new aws.iam.RolePolicy("ecsClusterInstanceRolePolicy", {
    role: instanceRole.id,  // (a)
    policy: JSON.stringify(instanceRolePolicyDoc),
});

let instanceProfile = new aws.iam.InstanceProfile("ecsClusterInstanceProfile", {
    role: instanceRole,  // (b)
});
```

At point (a), the role is specified using a string ARN.
https://github.com/pulumi/pulumi-aws/blob/8345675c388cc41797286ab10d53ddccae685928/pack/iam/rolePolicy.ts#L76
At point (b), the role must be specified as the `Role` object.
https://github.com/pulumi/pulumi-aws/blob/8345675c388cc41797286ab10d53ddccae685928/pack/iam/instanceProfile.ts#L90

I haven't looked into how the code is generated yet, but the underlying Terraform schema entries are both `schema.TypeString`:
https://github.com/terraform-providers/terraform-provider-aws/blob/90d698c66db80d676e2db197be97c1a9977e5393/aws/resource_aws_iam_instance_profile.go#L99
https://github.com/terraform-providers/terraform-provider-aws/blob/90d698c66db80d676e2db197be97c1a9977e5393/aws/resource_aws_iam_role_policy.go#L50

**Comment by jen20** [comment]
A similar issue for `RolePolicy` was fixed in #278. This shouldn't be too hard to go through and fix in the same manner without a breaking change using the same pattern.

**Comment by lukehoban** [comment]
Fixed by #278.  Thanks @jen20.
